lib/remap-functions/src/parse_url.rs

```diff
@@ -52,10 +53,25 @@ impl Expression for ParseUrlFn {
         self.value
             .type_def(state)
             .into_fallible(true) // URL parsing error
+            .with_inner_type(inner_type_def())
             .with_constraint(value::Kind::Map)
     }
 }
 
+/// The type defs of the fields contained by the returned map.
+fn inner_type_def() -> InnerTypeDef {
+    InnerTypeDef::Map(remap::type_def_map![
+        "scheme": TypeDef::new_with_kind(Kind::Bytes),
+        "username": TypeDef::new_with_kind(Kind::Bytes),
+        "password": TypeDef::new_with_kind(Kind::Bytes),
+        "path": TypeDef::new_with_kind(Kind::Bytes),
+        "host": TypeDef::new_with_kind(Kind::Bytes),
+        "port": TypeDef::new_with_kind(Kind::Bytes),
+        "fragment": TypeDef::new_with_kind(Kind::Bytes | Kind::Null),
```

Comment:
Good catch. Port can also be null. `username` and `password` seem to be empty strings rather than null.

```

$ parse_url!("http://onk.com")
{ "fragment": null, "host": "onk.com", "password": "", "path": "/", "port": null, "query": {  }, "scheme": "http", "username": "" }
```

Reply:
Correct. This is [in accordance with the spec](https://url.spec.whatwg.org/#url-representation) (I believe it's for backward compatibility reasons).